packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -79,49 +78,42 @@ export const useGiftCards = props => {
         }
     }, [cartId, getAppliedCards]);
 
-    // Submit the form after the apply or check balance actions are taken.
-    useEffect(() => {
-        if (
-            mostRecentAction === actions.APPLY ||
-            mostRecentAction === actions.CHECK_BALANCE
-        ) {
-            if (formApi) {
-                formApi.submitForm();
-            }
-        }
-    }, [formApi, mostRecentAction]);
-
     /*
      * useCallback hooks.
      */
-    const applyGiftCard = useCallback(() => {
-        // Ensure the mostRecentAction is APPLY before submitting.
-        if (mostRecentAction === actions.APPLY) {
-            if (formApi) {
-                formApi.submitForm();
-            }
-        } else {
-            // A useEffect will take care of submitting once this async
-            // operation finishes.
-            setMostRecentAction(actions.APPLY);
-        }
-    }, [formApi, mostRecentAction]);
+    const applyGiftCard = useCallback(async () => {
+        setMostRecentAction(actions.APPLY);
 
-    const checkGiftCardBalance = useCallback(() => {
-        // Ensure the mostRecentAction is CHECK_BALANCE before submitting.
-        if (mostRecentAction === actions.CHECK_BALANCE) {
-            if (formApi) {
-                formApi.submitForm();
+        const giftCardCode = formApi.getValue('card');
+
+        await applyCard({
+            variables: {
+                cartId,
+                giftCardCode
             }
-        } else {
-            // A useEffect will take care of submitting once this async
-            // operation finishes.
-            setMostRecentAction(actions.CHECK_BALANCE);
-        }
-    }, [formApi, mostRecentAction]);
+        });
+
+        // Clear the input form after successful apply.
+        formApi.reset();
+    }, [formApi, applyCard, cartId]);
+
+    const checkGiftCardBalance = useCallback(async () => {
```

Comment:
Doesn't look like this needs to be a Promise anymore.

Reply:
Great catch, thank you!